go/libraries/doltcore/schema/col_coll.go

```diff
@@ -143,6 +143,29 @@ func (cc *ColCollection) Append(cols ...Column) *ColCollection {
 	return NewColCollection(allCols...)
 }
 
+// IndexOf returns the index of the column with the name given (case-insensitive) or -1 if it's not found
+func (cc *ColCollection) IndexOf(colName string) (int, error) {
+	idx := -1
```

Comment:
in the past you've recommended leaning on the `NameToCol` and `TagToIndex` maps

Reply:
Those both use case-sensitive match unfortunately, makes them more trouble than they're worth from the SQL layer

Reply:
you added `LowerNameToCol` also